libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioAdapter.cs

```diff
@@ -0,0 +1,243 @@
+            {
+                throw new Exception("AuthToken is a required part of the configuration.");
+            }
+
+            TwilioClient.Init(_options.AccountSid, _options.AuthToken);
+        }
+
+        /// <summary>
+        /// Standard BotBuilder adapter method to send a message from the bot to the messaging API.
+        /// </summary>
+        /// <param name="turnContext">A TurnContext representing the current incoming message and environment.</param>
+        /// <param name="activities">An array of outgoing activities to be sent back to the messaging API.</param>
+        /// <param name="cancellationToken">A cancellation token for the task.</param>
+        /// <returns>A resource response.</returns>
+        public override async Task<ResourceResponse[]> SendActivitiesAsync(ITurnContext turnContext, Activity[] activities, CancellationToken cancellationToken)
+        {
+            var responses = new List<ResourceResponse>();
+            for (var i = 0; i < activities.Length; i++)
+            {
+                var activity = activities[i];
+                if (activity.Type == ActivityTypes.Message)
+                {
+                    var messageOptions = ActivityToTwilio(activity);
+
+                    var res = await MessageResource.CreateAsync(messageOptions);
+
+                    var response = new ResourceResponse()
+                    {
+                        Id = res.Sid,
+                    };
+
+                    responses.Add(response);
+                }
+                else
+                {
+                    // TODO: log error: unknown message type
+                }
+            }
+
+            return responses.ToArray();
+        }
+
+        /// <summary>
+        /// Accept an incoming webhook request and convert it into a TurnContext which can be processed by the bot's logic.
+        /// </summary>
+        /// <param name="request">A request object from Restify or Express.</param>
+        /// <param name="response">A response object from Restify or Express.</param>
+        /// <param name="bot">A bot with logic function in the form `async(context) => { ... }`.</param>
+        /// <param name="cancellationToken">A cancellation token for the task.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task ProcessAsync(HttpRequest request, HttpResponse response, IBot bot, CancellationToken cancellationToken = default)
+        {
+            response.StatusCode = 200;
+            await response.WriteAsync(string.Empty);
+
+            var twilioSignature = request.Headers["x-twilio-signature"];
+
+            var validationUrl = _options.ValidationUrl ?? (request.Headers["x-forwarded-proto"][0] ?? request.Protocol + "://" + request.Host + request.Path);
+
+            var requestValidator = new RequestValidator(_options.AuthToken);
+
+            var bodyStream = new StreamReader(request.Body);
```

Comment:
StreamReader is disposable, wrap this in a using block so it gets disposed

Reply:
Feedback applied.